### PR TITLE
Feature/775 lists and filters ux cleanup

### PR DESCRIPTION
**Pull request by amoedoamorim**
## Description

This ticket aims to fix the filters UX and make it consistent across core and custom lists.
The main part of this approach is to code around 3 different moments of a query:

- defaultQuery: which is the definition of a list
- stateQuery: which is whatever the user is composing
- appliedQuery: which is what is actually being displayed

This way we can code the buttons behavior like this:

canApply = stateQuery != appliedQuery
canSave = appliedQuery != defaultQuery
canReset = canApply || canSave

It also includes some cleaning up, removing dead code mentioning projects, adding string descriptions, normalizing the `page` prop, removing "update core lists" logic.

Channel filter now allows for selecting a specific tipline on the Tipline inbox filter.

Added SearchFieldDummy component for action-less filter fields.

References: CV2-775

## Type of change

- [x] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Tested manually and with integration tests that are already in place.

## Things to pay attention to during code review

- canReset, canApply & canSave logic as mentioned above
- merging defaultQuery and appliedQuery patterns

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [x] I have removed the /* eslint-disable @calm/react-intl/missing-attribute */ from any files I've worked on and added a `description` prop to all `<FormattedMessage />` components that were missing it.
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Comment by amoedoamorim** [comment]
@caiosba can you plz review this as Darius is OOO?

File: src/app/components/search/SaveList.js [hunk 1]
```diff
@@ -184,27 +154,9 @@ const SaveList = ({
...
-      if (objectType === 'tipline-inbox') {
-        input.tipline_inbox_filters = JSON.stringify(query);
-      } else if (objectType === 'suggested-matches') {
-        input.suggested_matches_filters = JSON.stringify(query);
-      }
-    } else {
-      input.filters = JSON.stringify(queryToBeSaved);
-    }
+    const input = {
+      filters: JSON.stringify({ ...query }),
```
Comment (caiosba) [inline src/app/components/search/SaveList.js:158]: Confirm that the default filters are included in this `query`.

File: src/app/components/search/SaveList.js [hunk 2]
```diff
@@ -274,16 +223,14 @@ const SaveList = ({
   };
 
   const handleClick = () => {
-    // From the "All Items" page and unmatched media page, we can just create a new list
-    if (objectType === 'all-items' || objectType === 'unmatched-media' || objectType === 'published') {
+    // From these pages we can just create a new list
```
Comment (caiosba) [inline src/app/components/search/SaveList.js:226]: Suggestion: Add a `FIXME` to declare this at a higher level where we can reuse.

File: src/app/components/search/SearchFields/SearchFieldDummy.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  onRemove,
+  version,
+}) => (
+  <RemoveableWrapper icon={icon} readOnly={readOnly} onRemove={onRemove} key={version}>
+    <Box px={0.5} height={4.5} display="flex" alignItems="center" whiteSpace="nowrap">
+      {label}
+    </Box>
+  </RemoveableWrapper>
+);
+
```
Comment (caiosba) [inline src/app/components/search/SearchFields/SearchFieldDummy.js:18]: Non-blocker: Add `PropTypes`

File: src/app/components/search/SearchFields/index.js [hunk 4]
```diff
@@ -163,91 +165,92 @@ const SearchFields = ({
...
   const handleRemoveField = (field) => {
-    const newQuery = { ...query };
+    const newQuery = { ...stateQuery };
     delete newQuery[field];
-    setQuery(newQuery);
+    setStateQuery(newQuery);
   };
 
-  const filterIsActive = () => Object.keys(query).filter(k => k !== 'keyword').length > 0;
+  // const filterIsActive = () => Object.keys(query).filter(k => k !== 'keyword').length > 0;
```
Comment (caiosba) [inline src/app/components/search/SearchFields/index.js:196]: As per our pair-review, this line needs to be deleted.

File: src/app/components/search/SearchFields/index.js
Comment (caiosba) [inline src/app/components/search/SearchFields/index.js:1]: Neat!

File: src/app/components/team/SuggestedMatches.js [hunk 5]
```diff
@@ -27,17 +27,18 @@ const SuggestedMatches = ({ routeParams }) => (
       render={({ error, props }) => {
         if (!error && props) {
           const { team } = props;
+          // Should we discard savedQuery already?
```
Comment (caiosba) [inline src/app/components/team/SuggestedMatches.js:30]: I think we can get rid of `savedQuery` already.

File: src/app/components/team/TiplineInbox.js [hunk 6]
```diff
@@ -30,32 +28,28 @@ const TiplineInbox = ({ routeParams }) => (
         if (!error && props) {
           const { team } = props;
           const defaultStatusId = team.verification_statuses.default;
+          // Should we discard savedQuery already?
```
Comment (caiosba) [inline src/app/components/team/TiplineInbox.js:31]: Yes, we can get rid of `savedQuery` already.

File: src/app/components/team/UnmatchedMedia.js [hunk 7]
```diff
@@ -6,12 +6,16 @@ import Search from '../search/Search';
 import UnmatchedIcon from '../../icons/unmatched.svg';
 
 const UnmatchedMedia = ({ routeParams }) => {
-  const query = {
-    unmatched: [1],
+  const defaultQuery = {
+    unmatched: ['1'],
```
Comment (caiosba) [inline src/app/components/team/UnmatchedMedia.js:10]: Review this change :)

**Review by caiosba — CHANGES_REQUESTED** [review]
Pair-reviewed synchronously with @amoedoamorim , great refactoring and clean up work involved here! :tada:

File: src/app/components/search/SaveList.js (hunk 1 above)
Comment (amoedoamorim) [reply src/app/components/search/SaveList.js:158]: Confirmed that filters from core lists are preserved when saving new lists

File: src/app/components/team/UnmatchedMedia.js (hunk 7 above)
Comment (amoedoamorim) [reply src/app/components/team/UnmatchedMedia.js:10]: image

This was showing in the stateQuery as a string anyway. I edited here to match so the logic for the display of save|reset|apply buttons would work as expected

Comment (amoedoamorim) [reply src/app/components/team/UnmatchedMedia.js:10]: The image above is the behavior when simply opening the Unmatched list, with the change reverted. Reset and apply should not appear.

File: src/app/components/search/SaveList.js (hunk 2 above)
Comment (amoedoamorim) [reply src/app/components/search/SaveList.js:227]: Done

File: src/app/components/search/SearchFields/index.js (hunk 4 above)
Comment (amoedoamorim) [reply src/app/components/search/SearchFields/index.js]: Done

File: src/app/components/team/SuggestedMatches.js (hunk 5 above)
Comment (amoedoamorim) [reply src/app/components/team/SuggestedMatches.js]: Done

File: src/app/components/team/TiplineInbox.js (hunk 6 above)
Comment (amoedoamorim) [reply src/app/components/team/TiplineInbox.js]: Done

File: src/app/components/search/SearchFields/SearchFieldDummy.js (hunk 3 above)
Comment (amoedoamorim) [reply src/app/components/search/SearchFields/SearchFieldDummy.js:19]: Done

**Review by caiosba — APPROVED** [review]
Thanks for the changes - I also tested locally two things for each core list:

- Default filters are displayed as read-only
- Default filters are preserved when saving new custom lists

Core lists that I tested:

- Tipline inbox
- Imported fact-checks
- Suggestions
- Unmatched
- Published

:tada:

**Pull request merged**